src/Model/Policy.php

```diff
@@ -535,6 +547,36 @@ public function setFieldIndex(string $ptype, string $field, int $index): void
         $assertion->fieldIndexMap[$field] = $index;
     }
 
+    /**
+     * Sets the current logger.
+     *
+     * @param Logger $logger
+     *
+     * @return void
+     */
+    public function setLogger(Logger $logger): void
+    {
+        foreach ($this->items as $sec => $astMap) {
+            foreach ($astMap as $ast) {
+                $ast->setLogger($logger);
+            }
+        }
+
+        $loggerAst = new Assertion();
+        $loggerAst->logger = $logger;
+        $this->items['logger']['logger'] = $loggerAst;
+    }
+
+    /**
+     * Returns the current logger.
+     *
+     * @return Logger
+     */
+    public function getLogger(): Logger
+    {
+        return $this->items['logger']['logger']->logger;
+    }
+
```

Comment:
Putting logger in item array is too rigid. `Model` is a class, not a `Map` in Go, please set a separate `logger` property.